src/ig-template/features/settings/Setting.ts

```diff
@@ -14,6 +15,8 @@ export abstract class Setting {
 
     requirement: Requirement;
 
+    protected _onChange = new SimpleEventDispatcher<SettingsValue[]>();
```

Comment:
I suggest using the regular `EventDispatcher` which takes 2 arguments instead of an array with a fixed length

Reply:
The `EventDispatcher` uses a sender and arguments no? I'm not entirely sure what you mean since it would just be `EventDispatcher<Setting, number[]>`.

Reply:
No the first argument doesn't have to be the sender

```ts
protected _onChange = new SimpleEventDispatcher<SettingsValue, SettingsValue>();
```
And invoke with
```ts
_onChange.dispatch(prevValue, newValue)
```
Instead of having an array with length 2